graph-http/src/intoresponse.rs

```diff
@@ -167,9 +173,9 @@ where
 }
 
 impl<'a, T> IntoResponseBlocking<'a, T> {
-    pub fn json<U>(self) -> GraphResult<GraphResponse<U>>
+    pub fn json<U, V>(self) -> GraphResult<GraphResponse<U>>
     where
-        for<'de> U: serde::Deserialize<'de>,
+        for<'de> U: serde::Deserialize<'de> + ODataNextLink<V>,
```

Comment:
We are gonna need a better solution then requiring every caller to implement ODataNextLink<V> just to call the json method. Yes the trait is implemented for serde_json::Value but not for user defined types. That would reduce functionality instead of increasing it and should probably be a different method.

Reply:
I ran into this problem when updating doc and example. 
That's why I added an impl in the proc macro for the singular type but I agree that we could do something better.

I wanted to implement it for all T but it's not possible unfortunately.

Reply:
Im struggling to figure out what the V in ODataNextLink<V> does in this situation? I dont see `Vec<V>` returned in the response so I am wondering what its use case is. Also, shouldnt this return the `json_value Vec<V>` since that includes all of the requests.?

Reply:
V is the type defined by the user, that is included in U as a Vec<V>. Unfortunately I didn't find a way to express that relationship for the compiler without a templated trait.


```
if let Some(json_value) = json.value() {

                json_value.append(&mut values);

            }
```
This part takes care of putting back all the requests data into the first request.